Libraries/LibWeb/MediaCapabilitiesAPI/MediaCapabilities.cpp

```diff
@@ -29,4 +127,120 @@ void MediaCapabilities::initialize(JS::Realm& realm)
     WEB_SET_PROTOTYPE_FOR_INTERFACE(MediaCapabilities);
 }
 
+// https://w3c.github.io/media-capabilities/#queue-a-media-capabilities-task
+void queue_a_media_capabilities_task(JS::VM& vm, Function<void()> steps)
+{
+    // When an algorithm queues a Media Capabilities task T, the user agent MUST queue a global task T on the
+    // media capabilities task source using the global object of the the current realm record.
+    queue_global_task(HTML::Task::Source::MediaCapabilities, vm.current_realm()->global_object(), GC::create_function(vm.current_realm()->heap(), move(steps)));
+}
+
+// https://w3c.github.io/media-capabilities/#dom-mediacapabilities-decodinginfo
+GC::Ref<WebIDL::Promise> MediaCapabilities::decoding_info(MediaDecodingConfiguration const& configuration)
+{
+    auto& realm = this->realm();
+    // The decodingInfo() method MUST run the following steps:
+
+    // 1. If configuration is not a valid MediaDecodingConfiguration, return a Promise rejected with a newly created
+    //    TypeError.
+    if (!configuration.is_valid_media_decoding_configuration()) {
+        return WebIDL::create_rejected_promise_from_exception(realm, vm().throw_completion<JS::TypeError>("The given configuration is not a valid MediaDecodingConfiguration"sv));
+    }
+
+    // 2. If configuration.keySystemConfiguration exists, run the following substeps:
+    // FIXME: Implement this.
+
+    // 3. Let p be a new Promise.
+    auto p = WebIDL::create_promise(realm);
+
+    // 4. Run the following steps in parallel:
+    auto& vm = this->vm();
+    Platform::EventLoopPlugin::the().deferred_invoke(GC::create_function(realm.heap(), [&vm, &realm, p, configuration]() mutable {
+        HTML::TemporaryExecutionContext context(realm, HTML::TemporaryExecutionContext::CallbacksEnabled::Yes);
```

Comment:
technically *this* context doesn't need callbacks enabled. only the one that interacts with promises does.

Reply:
Alright, done.